float32 error_pos
float32 error_att